### PR TITLE
chore: nix and rust update

### PR DESCRIPTION
**Pull request by tiagolobocastro**
chore(bdd): refactor code to support pytest-bdd 6

---

fix(nvme-list): support listings with no controller

The device list seems to differ locally and CI, possibly due to kernel differences
See the example snippet for some details
5.10.52 NixOS SMP Tue Jul 20 14:05:59 UTC 2021 x86_64 GNU/Linux
  Controllers:
  - Controller: nvme2
    Namespaces:
    - NameSpace: nvme2c2n1
      NSID: 1
  Namespaces:
  - NameSpace: nvme2n1
    NSID: 1
 VS
5.18.10 NixOS SMP PREEMPT_DYNAMIC Thu Jul 7 15:55:01 UTC 2022 x86_64 GNU/Linux
  Controllers:
  - Controller: nvme2
    Namespaces:
    - NameSpace: nvme2n1
      NSID: 1

---

chore(niv): update nixpkgs to release-22.05

This bumps the stable rust to 1.62.0
Set the nightly to 1.63.0/2022-06-22

---

chore(nix-shell): allow selecting a rust profile (stable/nightly)

Example:
$> nix-shell --run zsh --argstr rust-profile nightly

---

fix(nix): support the latest git versions and nix

Use GIT_DIR to allow for the version derivation with a different user.
Allow use of the nix cmdline.
Replace libudev with udev (rename).

---

chore: update dependencies to latest develop

---

fix: address new clippy warnings

### CODE REVIEW

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

File: nix/lib/rust.nix [hunk 1]
```diff
@@ -7,7 +7,7 @@ let
 in
 rec {
   rust_default = { override ? { } }: rec {
-    nightly_pkg = pkgs.rust-bin.nightly."2021-11-22";
+    nightly_pkg = pkgs.rust-bin.nightly."2022-06-22";
```
Comment (Abhinandan-Purkait) [inline nix/lib/rust.nix:10]: Is it worth to make this change for data-plane as well? Having same versions for both would ease development.

File: tests/bdd/features/csi/node/test_parameters.py [hunk 2]
```diff
@@ -102,7 +113,7 @@ def staging_target_path():
 
 
 @pytest.fixture
-def start_csi_plugin(setup, io, staging_target_path):
+def start_csi_plugin(setup, io_queues, staging_target_path):
```
Comment (Abhinandan-Purkait) [inline tests/bdd/features/csi/node/test_parameters.py:116]: Thanks for this change.

File: nix/pkgs/control-plane/cargo-project.nix [hunk 3]
```diff
@@ -12,7 +12,7 @@
 , version
 , openapi-generator
 , which
-, libudev
+, udev
```
Comment (Abhinandan-Purkait) [inline nix/pkgs/control-plane/cargo-project.nix:15]: Question: Why is this change needed?

**Review by Abhinandan-Purkait — APPROVED** [review]

File: nix/lib/rust.nix (hunk 1 above)
Comment (tiagolobocastro) [reply nix/lib/rust.nix:10]: Yep, it'd be good to get the latest clippy suggestions.

File: nix/pkgs/control-plane/cargo-project.nix (hunk 3 above)
Comment (tiagolobocastro) [reply nix/pkgs/control-plane/cargo-project.nix:15]: The library was renamed

**Review by paulyoong — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge
